#197. notification re-transmittion added. PDRs below 1 work

Raa37 opened the pull request.

twatteyne commented on DotBot.py:
```diff
@@ -5,6 +5,7 @@
 # local
 import SimEngine
 import Wireless
+import Orchestrator
```
big big big no-no, DotBot cannot konw about Orchestrator

twatteyne commented on DotBot.py:
```diff
@@ -18,6 +19,8 @@ class DotBot(Wireless.WirelessDevice):
     A single DotBot.
     '''
 
+    COMM_DOWNSTREAM_PERIOD_S      = 1
```
certainly not the right name, DotBot doesn't gnerate any downstream data...

twatteyne commented on DotBot.py:
```diff
@@ -36,6 +39,9 @@ def __init__(self, dotBotId, x, y, floorplan):
         self.tsMovementStart      = None
         self.tsMovementStop       = None
+        # sequence numbers (to filter out duplicate commands and notifications)
+        self.seqNumMovement       = None
```
what does movement mean? Do you refer to commands (Orchestrator->DotBot) and notifications (DotBot->orchestrator)?

twatteyne commented on DotBot.py:
```diff
@@ -162,6 +176,13 @@ def _transmit(self):
             sender      = self,
         )
 
+        # schedule re-transmit
```
"schedule retransmission"

twatteyne commented on DotBot.py:
```diff
@@ -162,6 +176,13 @@ def _transmit(self):
 
+        # schedule re-transmit
+        self.simEngine.schedule(
+            ts=self.simEngine.currentTime() + self.COMM_DOWNSTREAM_PERIOD_S,
```
indent

twatteyne reviewed (changes requested):
small changes, should be easy to fix. PLease add a comment indicating what you did for testing

Raa37 commented:
Testing method:
1. changed PDR to 0.8 and saw how non of the cells were placed correctly anymore.
2. added retransmission and sequence numbers and checked that simulation ran to completion, placing cells in correct positions.
3. I then ran simulation with PDR = 0.3 to check that time to completion increases yet still completes. and indeed it took half an hou8r more to complete (in simulation time)

twatteyne commented on DotBot.py:
```diff
@@ -40,7 +39,7 @@ def __init__(self, dotBotId, x, y, floorplan):
         self.tsMovementStop       = None
         # sequence numbers (to filter out duplicate commands and notifications)
-        self.seqNumMovement       = None
+        self.seqNumCommand       = None
```
Indent

twatteyne commented on DotBot.py:
```diff
@@ -19,7 +18,7 @@ class DotBot(Wireless.WirelessDevice):
     '''
 
-    COMM_DOWNSTREAM_PERIOD_S      = 1
+    RETRY_TRANSMIT_PERIOD_S      = 1
```
RETRY_TIMEOUT_S

twatteyne reviewed (changes requested):
Once fixed, please merge
